README.md

```diff
@@ -42,11 +42,24 @@ Change history can be found in [`HISTORY.md`](https://github.com/mongodb/node-mo
 
 ### Compatibility
 
-For version compatibility matrices, please refer to the following links:
+For server and runtime version compatibility matrices, please refer to the following links:
 
 - [MongoDB](https://www.mongodb.com/docs/drivers/node/current/compatibility/#mongodb-compatibility)
 - [NodeJS](https://www.mongodb.com/docs/drivers/node/current/compatibility/#language-compatibility)
 
+#### Component Support Matrix
+
+The following table describes add-on compponent version compatibility for the Node.js driver. Only packages with versions in these supported ranges are stable when used in combination.
+
+| Component                 | mongodb@3.x | mongodb@4.x        | mongodb@5.x        | mongodb@6.x |
+| ------------------------- | ----------- | ------------------ | ------------------ | ----------- |
+| bson                      | ^1.0.0      | ^2.0.0             | ^5.0.0             | ^6.0.0      |
+| bson-ext                  | ^1.0.0      | ^2.0.0             | N/A                | N/A         |
+| kerberos                  | ^1.0.0      | ^1.0.0 \|\| ^2.0.0 | ^1.0.0 \|\| ^2.0.0 | ^2.0.0      |
+| mongodb-client-encryption | ^1.0.0      | ^1.0.0 \|\| ^2.0.0 | ^2.3.0             | ^6.0.0      |
+| mongodb-legacy            | N/A         | ^4.0.0             | ^5.0.0             | ^6.0.0      |
+| \@mongodb-js/zstd         | N/A         | ^1.0.0             | ^1.0.0             | ^1.1.0      |
```

Comment:
Can we make the components link to either their GH project pages or their npm pages?

Reply:
@alexbevi you didn't see the todo in my PR description :) I left this PR in draft while it wasn't ready; should be good now